src/clickup/service.ts

```diff
@@ -0,0 +1,108 @@
+import { IMerjoonProjects, IMerjoonService, IMerjoonTasks, IMerjoonUsers } from '../common/types';
+import {
+  IClickUpMember,
+  IClickUpList,
+  IClickUpItem,
+  IClickUpTeam,
+} from './types';
+import { ClickUpTransformer } from './transformer';
+import { ClickUpApi } from './api';
+
+export class ClickUpService implements IMerjoonService {
+
+  protected teamIds: string[] | undefined;
+  protected spaceIds: string[] | undefined;
+  protected folderIds: string[] | undefined;
+  protected listIds: string[] | undefined;
+  
+  constructor(public readonly api: ClickUpApi, public readonly transformer: ClickUpTransformer) {
+  }
+
+  static mapIds(items: IClickUpItem[]) {
+    return items.map((item: IClickUpItem) => item.id);
+  }
+
+  protected async getTeams() {
+    return this.api.getTeams();
+  }
+
+  protected async getSpaces() {
+    if (!this.teamIds) {
+      throw new Error('Team IDs not found');
+    }
+    return Promise.all(this.teamIds.map((id) => this.api.getTeamSpaces(id)));
```

Comment:
const items = await Promise.all(this.teamIds.map((id) => this.api.getTeamSpaces(id)));
return items.flat()